azafea/event_processors/endless/metrics/v2/model/__init__.py

```diff
@@ -775,10 +776,29 @@ class LinuxPackageOpened(SingularEvent):
     #: argv of the launched Windows application
     argv = Column(ARRAY(Unicode, dimensions=1), nullable=False)
 
+    def _parse_payload(self, maybe_payload: GLib.Variant) -> Dict[str, Any]:
+        payload = maybe_payload.get_maybe()
+
+        if payload is None:
+            raise EmptyPayloadError(f'Metric event {self.__event_uuid__} needs a '
+                                    f'{self.__payload_type__} payload, but got none')
+
+        payload = get_variant(payload)
+        payload_type = payload.get_type_string()
+
+        if payload_type != self.__payload_type__ and payload_type != 's':
+            raise WrongPayloadError(f'Metric event {self.__event_uuid__} needs a '
+                                    f'{self.__payload_type__} payload, but got '
+                                    f'{payload} ({payload_type})')
+        return self._get_fields_from_payload(payload)
+
     @staticmethod
     def _get_fields_from_payload(payload: GLib.Variant) -> Dict[str, Any]:
+        value = payload.unpack()
+        if type(value) == str:
```

Comment:
I don't think it makes any real difference, but I prefer `isinstance(value, str)`.